components/Tabs/TabContext.tsx

```diff
@@ -0,0 +1,50 @@
+import { createContext, useState } from "react";
+import { createHash } from "crypto";
+import type { BinaryToTextEncoding } from "crypto";
+import { DataTab } from "./types";
+import type { ReactNode } from "react";
+
+export interface TabContextProps {
+  getSelectedLabel: (tabs: Array<string>) => string;
+  setSelectedLabel: (tabs: Array<string>, selected: string) => void;
+}
+
+export const TabContext = createContext<TabContextProps>({
+  getSelectedLabel: (tabs: Array<string>): string => {
+    return "";
+  },
+  setSelectedLabel: (tabs: Array<string>, selected: string) => {
+    return;
+  },
+});
+
+const labelHashKey = (labels: Array<string>): string => {
+  const labelsCopy = [...labels];
+  labelsCopy.sort();
+  const hash = createHash("sha256");
+  hash.update(labelsCopy.join(""));
+  return hash.digest("utf8" as BinaryToTextEncoding);
+};
+
```

Comment:
I felt weird about having an unbounded key. I guess we'll never have hundreds of labels or anything, so this won't be a big deal. I've changed this to return the joined labels instead of hasing.